Fix typo introduced in 5031301

MorganJamesSmith opened the pull request:
Emacs is very unhappy about this complaining about "void-variable (-*-)" for a very valid reason.  I looked through the rest of commit 5031301 and this seems to be the only typo.

I'm going to mention @ErikSchierboom and @fapdash since they where responsible for this change.  Not to be mean, just to let them know.  Everyone makes mistakes :)

Thanks everyone for making this course!  I can't wait to get past the first exercise :P

fapdash commented:
@MorganJamesSmith Oops, yeah I automated the change with a bash script and didn't check thoroughly enough. Thank you for providing a fix. :)

>  I can't wait to get past the first exercise :P

You should be able to submit a valid solution right now, no? `hello-world.el` is just a template that you can change however you like.

ErikSchierboom commented on exercises/practice/hello-world/hello-world.el:
```diff
@@ -1,4 +1,9 @@
-(defun hello ()  -*- lexical-binding: t; -*-
+;;; hello-world.el --- Hello World exercise (exercism)  -*- lexical-binding: t; -*-
```
What is the minimal amount of code you can use here?

MorganJamesSmith replied:
I'm not sure I understand the question

SleeplessByte replied:
The function declaration was there for a reason, in order to make the amount of steps the user needs to take as minimal as possible.

We recommend even prefilling the return value with "Goodbye, Mars!" And have the user change that to Hello, World! as this exercise is meant to get your computer in shape, not to be an actual challenge.

ErikSchierboom replied:
@SleeplessByte I think the function declaration is there, but just in another part of the file.

SleeplessByte replied:
Ah. Doesn't change my other comment then 😁😁

ErikSchierboom replied:
You're referring to addings more comments to explain what's happening?

fapdash replied:
Maybe something like this? (mostly quoting from the Emacs Manual)

```elisp
;;; hello-world.el --- Hello World exercise (exercism)  -*- lexical-binding: t; -*-

;;; Commentary:

;;; Emacs has conventions for using special comments in Lisp libraries that give structured information about the library.
;;; Using a standard format for these items makes it easier for tools (and people) to extract the relevant information.
;;; For more information, see
;;; https://www.gnu.org/software/emacs/manual/html_node/elisp/Library-Headers.html

;;; Code:
```

MorganJamesSmith commented:
@ErikSchierboom If you take a look at the other exercises here they all share the boiler plate comments I added except for this one.  Emacs will throw warnings at the user if it does not detect this boiler plate.

So yes, I believe every line I added is indeed necessary.

MorganJamesSmith commented on exercises/practice/hello-world/hello-world.el:
```diff
@@ -1,4 +1,9 @@
   "Goodbye, Mars!")
 
 (provide 'hello-world)
+;;; hello-world.el ends here
```
Without this, Emacs will emit a warning to the user

MorganJamesSmith commented on exercises/practice/hello-world/hello-world.el:
```diff
@@ -1,4 +1,9 @@
-(defun hello ()  -*- lexical-binding: t; -*-
+;;; hello-world.el --- Hello World exercise (exercism)  -*- lexical-binding: t; -*-
+
+;;; Commentary:
```
Without the header and the "Commentary" line, Emacs will emit a warning to the user

SleeplessByte replied:
I think it makes sense to prefill the comment in hello world with something like:

The lines above start a comment block yadaydaydaya this is a requirement yadhaydyaydya to explain first time users why it's present in the first place.

SleeplessByte reviewed (changes requested):
See responses to comments

ErikSchierboom commented:
@SleeplessByte See https://github.com/exercism/emacs-lisp/pull/237#discussion_r1017467946

MorganJamesSmith commented:
Ok so we all seem to be mostly ok with adding boiler plate comments if it silences warnings.  It seems our main concern is confusing new users.

After looking through 4 other hello-world templates (Python, elixer, haxe, ruby) I have discovered that instructional comments are not present in those templates.  Looking at a few HELP.md's shows that exercism is more focused on offering problems, not documentation.  Correct me if this is wrong, I'm very new to exercism.  IMO it is always better to refer users to the official documentation.

ErikSchierboom commented:
> After looking through 4 other hello-world templates (Python, elixer, haxe, ruby) I have discovered that instructional comments are not present in those templates. Looking at a few HELP.md's shows that exercism is more focused on offering problems, not documentation. Correct me if this is wrong, I'm very new to exercism. IMO it is always better to refer users to the official documentation.

JavaScript has a very minor instructional comment: https://github.com/exercism/javascript/blob/main/exercises/practice/hello-world/hello-world.js, but it is not that common.

MorganJamesSmith commented:
The comment you pointed out doesn't actually explain any syntax in the file though.

I think my PR here succeeds in fixing the initial typo and should be merged.  The idea of adding extra instructions are beyond the scope of this PR but is a cool idea to think about.

@SleeplessByte said:
> this exercise is meant to get your computer in shape, not to be an actual challenge.

If this is true I would imagine figuring out where to find documentation is part of "Hello, World!" and if the documentation doesn't sufficiently explain the boilerplate, that is the fault of the documentation, not exercism

ErikSchierboom commented:
> I think my PR here succeeds in fixing the initial typo and should be merged. The idea of adding extra instructions are beyond the scope of this PR but is a cool idea to think about.

I agree with this.
